[Finishes #171293123] chancla

Adrian-de-Lucas-Gomez opened the pull request:
Ya está hecho y funcional. Cuando cojas la chancla se desactiva la colisión de la misma y se activa el trigger que al entrar en contacto con los otros jugadores les aplica el daño correspondiente.

Adrian-de-Lucas-Gomez committed:
Cambios a player data y metodo Accion

He hecho que cuando se coge un arma se guarde el componente health del juegador porque a la hora de atacar con la chancla se necesita saber el nivel de vida del jugador que la usa para saber cuanto daño va a infringir al contrincante.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Merge branch 'Pelota-Saltarina' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Mas intentos pero aun sin éxito

Se ha creado una clase que deriva de Weapon pero especifica para las armas de Melee para tratar de suplir las necesidades que estas van a tener, si es que se hace alguna mas a parte de la chancla claro.

Adrian-de-Lucas-Gomez committed:
Está tan cerca y a la vez tan lejos

Está a punto de caramelo. Lo he hecho por sensors pero tengo que ver como detectarlo en condiciones para que no mueva al jugador

Adrian-de-Lucas-Gomez committed:
Chancla ya funcional pero a veces funky

Se le ha puesto un trigger algo mas grande que la caja de colision de la propia chancla. Al entrar en jugador enemigo en ese área se comprueba si la chancla esta en modo ataque y si es así se mira el daño que debe hacer y se lo aplica al enemigo.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Un par de ajustes

Al hacer merge daba un error pero ya lo he solucionado quitando unas cosas

Adrian-de-Lucas-Gomez committed:
Chancla acabada

Done Done

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

AdriAlvarez00 reviewed (approved):
No pises a tinky, que lleva cholas

EstebenR reviewed (changes requested):
¿Por que se ha borrado lo de llamar a action en Weapon.cpp?

Also mergear el master pls

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Listo

Ya esta mergeado on el master actualizado

Adrian-de-Lucas-Gomez commented:
> 
> 
> ¿Por que se ha borrado lo de llamar a action en Weapon.cpp?
> 
> Also mergear el master pls

Ya está. No se llama a Action ya al pulsar el botón ya que el modo de ataque de la chancla se pone nada mas cogerla por lo que Action ya unicamente calcula el daño de debe de hacer el jugador al golpear.

EstebenR committed:
Update Health.cpp

EstebenR committed:
Cambiados los includes de sitio

EstebenR committed:
Arreglados filters

Adrian-de-Lucas-Gomez committed:
Ya esta listo

Funciona mejor que nunca

Adrian-de-Lucas-Gomez committed:
Cooldown arma melee

Al pulsar X el jugador activa el poder destructivo de la chancla que durará tantos updates como marque la constante WEAPON_MELEE_TIME la cual esta en el json donde están las otras constantes.

EstebenR reviewed (approved):
Bene

mmur07 reviewed (changes requested):
La chancla tiene que hacer knockback

Adrian-de-Lucas-Gomez committed:
Chancla con Knockback

Funciona igual que en el commit anterior (Se cambiará al modelo propuesto cuando se pueda) solo que ahora se aplica un knockback al jugador en la direccion contraria a donde impacta con la chancla.

mmur07 commented:
> 
> 
> La chancla tiene que hacer knockback

Okay, queda bien con el knockback alto. Igual quedaría menos wonky si la dirección del impulso fuera desde el jugador que tiene la chancla hacia el enemigo en vez de partir desde la chancla hacia el enemigo. También habría que especificar exactamente cómo queremos que funcionen las armas (si con el swing tuyo o con animación fija). La culpa recae en mi que tampoco lo especificamos en el GDD asi que sorry por mi parte. Ya si eso lo podemos comentar mañana en el scr00m 👌

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Chancla con rango

Falta eliminar a los enemigos de la lista cuando salen del rango, pero es cosa de que tengo que mirarlo mas en detalle

Adrian-de-Lucas-Gomez committed:
Update constants.json

Adrian-de-Lucas-Gomez committed:
Chancla Redux Deluxe Edition Director's Cut

La chancla ahora funciona de forma parecida al router metiendo en un vector a los jugadores que se encuentran en rango. Al pulsar el boton de atacar se recorre ese vector y se aplica el daño e impulso correspondiente y se activa el cooldown (Puesto en una variable). Ahora al coger la chancla la vemos cuando podemos atacar y desaparece tras efectuar un ataque.

EstebenR reviewed (changes requested):
Si ataco con la chancla y acto seguido la tiro salta excepcion

Adrian-de-Lucas-Gomez committed:
Excepcion arreglada

Pasaba que trataba de borrar una fixture en la posicion 2 pero cuando se crea una nueva es en la posicion siguente. Lo he echo que vaya variando y ahora va de cine

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into #171293123-Chancla

Adrian-de-Lucas-Gomez committed:
Listo para meter en master

Se ha mergeado con master y se han solucionado los problemas tipicos

EstebenR reviewed (approved):
ya furula

mmur07 reviewed (approved):
te perd0no !1
